emulator.go

```diff
@@ -239,6 +241,13 @@ func (s *Server) CreateTask(ctx context.Context, in *tasks.CreateTaskRequest) (*
 
 	queueName := in.GetParent()
 	queue, ok := s.fetchQueue(queueName)
+
+	// If auto create queue is on we try to create queue if not existing.
+	if !ok && autoCreateQueueOnNewTask {
+		createInitialQueue(s, queueName)
```

Comment:
Using the initial queue method is probably not appropriate here as this will be called in 'normal operation' (not as an initialization). For example, it will display "creating initial queue" (not so relevant), and will panic if it encounters an error.

Reply:
I am not sure how to resolve this, I can create a new method but it will duplicate a large portion of the CreateQueue function. I can call CreateQueue function but it would mean create a tasks.CreateQueueRequest inside the CreateTask function and I am not sure it is the best approach.
It seems only the queueName and the parentName are used to create a queue so I could just extract the code of CreateQueue to a function use by both. Maybe you see a better approach ?

Reply:
Hi @JeanDaniel-Fischer, apologies for the delay in my response. Yes I see your point;
To understand what we should reuse, can I ask what you expect in terms of error handling? I.e. if the queue name does not meet the required format, do you expect an error of some sort like it would give you if you create the queue separately? Obviously the benefit of reusing `CreateQueue` is that you can reuse the error checking done there and the error state generated (if any).